internal/pkg/mining/block_generate.go

```diff
@@ -55,6 +55,9 @@ func (w *DefaultWorker) Generate(
 	mq := NewMessageQueue(pending)
 	candidateMsgs := orderMessageCandidates(mq.Drain())
 	candidateMsgs = w.filterPenalizableMessages(ctx, candidateMsgs)
+	if len(candidateMsgs) > block.BlockMessageLimit {
```

Comment:
We should pass the size limit as a parameter to Drain so that we don't take messages out of the pool and then drop them.  It's ok to merge as is, I will file an issue.

Reply:
I'm going to take a stab at doing this.  Will back out if it gets too crazy.